packages/client/hmi-client/src/components/workflow/ops/interventions/tera-interventions-drilldown.vue

```diff
@@ -0,0 +1,395 @@
+	const state = cloneDeep(props.node.state);
+	knobs.value.transientInterventionPolicy = cloneDeep(interventionPolicy);
+
+	const listOfPolicyIds: string[] = props.node.outputs.map((output) => output.value?.[0]);
+	// Update output port:
+	if (!interventionPolicy?.id) {
+		logger.error('Policy not found');
+		return;
+	}
+	// Check if this output already exists
+	if (listOfPolicyIds.includes(interventionPolicy.id)) {
+		// Select the existing output
+		const output = props.node.outputs.find((ele) => ele.value?.[0] === interventionPolicy.id);
+		emit('select-output', output?.id);
+	}
+	// If the output does not already exist
+	else {
+		// Append this config to the output.
+		state.transientInterventionPolicy = interventionPolicy;
+		emit('append-output', {
+			type: InterventionsOperation.outputs[0].type,
+			label: interventionPolicy.name,
+			value: interventionPolicy.id,
+			state
+		});
+	}
+	logger.success(`Policy applied ${interventionPolicy.name}`);
+};
+
+const fetchInterventionPolicies = async (modelId: string) => {
+	isFetchingPolicies.value = true;
+	interventionsPolicyList.value = await getInterventionPoliciesForModel(modelId);
+	isFetchingPolicies.value = false;
+};
+
+const onUpdate = (intervention: Intervention, index: number) => {
+	knobs.value.transientInterventionPolicy.interventions[index] = cloneDeep(intervention);
+};
+
+const onSelection = (id: string) => {
+	emit('select-output', id);
+};
+
+const onUsePolicy = (policy: InterventionPolicy) => {
+	confirm.require({
+		header: 'Are you sure you want to use this intervention policy?',
+		message: `All current interventions will be replaced with those in the selected policy, “${policy.name}” This action cannot be undone.`,
+		accept: () => applyInterventionPolicy(policy),
+		acceptLabel: 'Confirm',
+		rejectLabel: 'Cancel'
+	});
+};
+
+const onAddIntervention = () => {
+	// by default add the first parameter with a static intervention
+	const intervention: Intervention = {
+		name: 'New Intervention',
+		appliedTo: parameterOptions.value[0].value,
+		staticInterventions: [{ threshold: 0, value: 0 }],
+		dynamicInterventions: []
+	};
+	knobs.value.transientInterventionPolicy.interventions.push(intervention);
+};
+
+const onDeleteIntervention = (index: number) => {
+	knobs.value.transientInterventionPolicy.interventions.splice(index, 1);
+};
+
+const onChangeName = async (name: string) => {
+	if (!selectedPolicy.value) return;
+	selectedPolicy.value.name = name;
+	await updateInterventionPolicy(selectedPolicy.value);
+	updateNodeLabel(selectedOutputId.value, name);
+	initialize();
+};
+
+const onEditDescription = () => {
+	isEditingDescription.value = true;
+	newDescription.value = selectedPolicy.value?.description ?? '';
+};
+
+const onConfirmEditDescription = async () => {
+	if (!selectedPolicy.value) return;
+	selectedPolicy.value.description = newDescription.value;
+	isEditingDescription.value = false;
+	await updateInterventionPolicy(selectedPolicy.value);
+	initialize();
```

Comment:
Im not following why we are re-initializing on every description change

Reply:
see comment for name